common/djangoapps/student/views/management.py

```diff
@@ -403,6 +404,12 @@ def change_enrollment(request, check_access=True):
         if redirect_url:
             return HttpResponse(redirect_url)
 
+        if CourseEntitlement.check_for_existing_entitlement_and_enroll(
+            user=user,
+            course_run_key=course_id
+        ) is not None:
+            return HttpResponse(reverse('dashboard'))
```

Comment:
Could we redirect them directly into the course?

Reply:
When a user enrolls in a course currently they are redirected to the dashboard.  I am not sure we should change that in this PR. 

I will verify that is what is currently done.  Also I would want to loop in Deen as we discussed this behavior originally and decided on the Dashboard.

Reply:
NM, I was wrong here.  We send the user to the Courseware.  I will take a look at redirecting them in this case as well. Good catch. 👍

Reply:
Did we decide to keep it as dashboard overall or are you still looking into this? (bumping just cause this is marked as outdated in PR comments)

Reply:
I moved it in the code.  Take a look.